omod/src/main/java/org/openmrs/module/referenceapplication/page/controller/LoginPageController.java

```diff
@@ -110,11 +114,24 @@ public String get(PageModel model,
 			Context.removeProxyPrivilege(GET_LOCATIONS);
 		}
 
+		model.addAttribute("showSessionLocations", !isLocationUserPropertyAvailable(administrationService));
+		Object showLocation = pageRequest.getAttribute("showSessionLocations");
+		if(showLocation != null && showLocation.toString().equals("true")) {
+		    model.addAttribute("showSessionLocations", true);
+		}
 		model.addAttribute("lastSessionLocation", lastSessionLocation);
 
 		return null;
 	}
 
+	private boolean isLocationUserPropertyAvailable(AdministrationService administrationService) {
+		String locationUserPropertyName = administrationService.getGlobalProperty(ReferenceApplicationConstants.LOCATION_USER_PROPERTY_NAME);
+		if(StringUtils.isNotBlank(locationUserPropertyName) && !locationUserPropertyName.equals("false")) {
```

Comment:
I thought we agreed to simply use a valid user property name. Instead instead of true or false

Reply:
Yes , Whenever it is set, we know that it is true. If not set, we know it is false.